test/Serial_TC.cpp

```diff
@@ -0,0 +1,28 @@
+#include <Serial_TC.h>
+
+#include "ArduinoUnitTests.h"
+
+unittest(SerialTest) {
+  GodmodeState* state = GODMODE();
+  state->serialPort[0].dataIn = "";   // the queue of data waiting to be read
+  state->serialPort[0].dataOut = "";  // the history of data written
+
+  TC_Serial mySerial;
+
+  assertTrue(true);
+
+  mySerial.write('b');
+  assertEqual("", state->serialPort[0].dataIn);
+  assertEqual("b", state->serialPort[0].dataOut);
+
+  String test = "test:";
+  mySerial.print(test);
+  assertEqual("", state->serialPort[0].dataIn);
+  assertEqual("btest:\r\n", state->serialPort[0].dataOut);
+
+  mySerial.print(test, 3);
+  assertEqual("", state->serialPort[0].dataIn);
+  assertEqual("btest:\r\ntest:3\r\n", state->serialPort[0].dataOut);
+}
+
+unittest_main()
```

Comment:
We should have tests for each of the public API functions. It wouldn't take much to add a few more.